utils/config_file.hh

```diff
@@ -70,7 +70,10 @@ public:
               ///< "Used" options are added to the command line options,
               ///< and can be specified with command line.
         Unused, ///< an option inherited or deprecated, or not yet implemented.
-                ///< we just silently ignore these options at seeing them.
+                ///< We want to minimize their "visibility" from user. So
+                ///< despite that they are still accepted in the config file,
+                ///< they are not considered as valid options if specified
+                ///< using the command line anymore.
```

Comment:
It sounds like this "still works in scylla.yaml but not allowed in command line" is just a historic accident, is there any reason why we want to keep it this way?
Also, I think you need to clarify what's the difference between Unused and Deprecated in purpose (not in implementation) - you can't continue to say that "Unused" is for "deprecated" options, since we now have a separate "Deprecated". So, what do we continue to use "Unused" for? Maybe it's just for unimplemented Cassandra features? Or what?

Reply:
> It sounds like this "still works in scylla.yaml but not allowed in command line" is just a historic accident, is there any reason why we want to keep it this way? 

I think it's just the "historical accident" case.

> Also, I think you need to clarify what's the difference between Unused and Deprecated in purpose (not in implementation) - you can't continue to say that "Unused" is for "deprecated" options, since we now have a separate "Deprecated". So, what do we continue to use "Unused" for? Maybe it's just for unimplemented Cassandra features? Or what?

I think that the _deprecated_ word used in both `value_status::Invalid/Unused` enumerators comments refers to how we handled deprecation of config options before, i.e. moving them from e.g. `Used` to `Unused`. Now, that we have this new `Deprecated` enumerator, deprecation would happen by moving config options from `Used` to `Deprecated`, (then maybe from `Deprecated` to `Unused/Invalid`? Or removing altogether?). 
Would it be OK if I changed `Unused/Invalid` to say that they were historically used for deprecation?

Reply:
Yes (for the last question). Since the purpose of this patch is to fix the messy situation we're in, I think it should actually do that - make the situation less messy, not more messy. I think you should start by deciding what you **want** to happen, what will not be messy. Most importantly, please don't leave comments that keep saying that both Unused and Deprecated means deprecated today - because this will confuse future developers who won't understand which of the two to use. Maybe you can say something like "Unused used to mean ... but don't use it in new code, use Deprecated instead"? Or just get rid of the Unused option completely?

Reply:
I removed mentioning deprecation from Unused and Invalid, and to both appended:
>                 ///< (In the past used to deprecate an option,
>                 ///< now there's a separate Deprecated enumerator for that purpose)